docs/usage.rst

```diff
@@ -42,21 +46,57 @@ Cut-and-paste into your config file:
 
 .. code-block:: ini
 
-   yourapp.session_secret =
+   session.secret =
       840aaafdc36f067fbad9baf006efc0f672b86ab0dcb6a3e43ecc1f9d760915e5
 
-Parse the secret from the configuration, de-hexlify it, and use it to
-construct and register a session factory:
+Session management
+``````````````````
+If you have included pyramid_nacl_session in your Pyramid project's configuration
+as shown above then :func:`pyramid_nacl_session.session_factory_from_settings`
+is called automatically and you need do nothing else.
+Otherwise you will need to create a Pyramid 
+:term:`session factory` by adding a call to either the
+:func:`pyramid_nacl_session.EncryptedCookieSessionFactory` function or the
+:func:`pyramid_nacl_session.session_factory_from_settings` function in
+the configuration code of your Pyramid project's ``__init__.py`` file
+and subsequently register that
+session factory with Pyramid.  At that point, accessing
+``request.session`` will provide a Pyramid session using PyNaCl as a
+backend.
+
+:func:`pyramid_nacl_session.session_factory_from_settings` obtains session
+settings from the ``**settings`` dictionary passed to the
+Configurator.  It assumes that you've placed session configuration
+parameters prefixed with ``session.`` in your Pyramid application's
+``.ini`` file.  For example:
 
-.. code-block:: python
+.. code-block:: ini
 
-   import binascii
-   from pyramid.session import BaseCookieSessionFactory
-   from pyramid_nacl_session import EncryptedCookieSessionFactory
+   [app:myapp]
+   .. other settings ..
+   session.secret = 840aaafdc36f067fbad9baf006efc0f672b86ab0dcb6a3e43ecc1f9d760915e5
+   session.serializer = json
 
-   def includeme(config):
-       hex_secret = config.settings['yourapp.session_secret'].strip()
-       secret = binascii.unhexlify(hex_secret)
-       factory = EncryptedCookieSessionFactory(serializer)  # other config ad lib.
-       config.set_session_factory(factory)
+If your ``.ini`` file has such settings, you can use
+:func:`pyramid_nacl_session.session_factory_from_settings` in your
+application's configuration.  For example, let's assume this code is
+in the ``__init__.py`` of your Pyramid application that uses an
+``.ini`` file with the ``session.`` settings above to obtain its
+``**settings`` dictionary.
+
+.. code-block:: python
 
+   from pyramid_nacl_session import session_factory_from_settings
+   from pyramid.config import Configurator
+
+   def app(global_config, **settings):
+       """ This function returns a WSGI application.
+       
+       It is usually called by the PasteDeploy framework during 
+       ``paster serve``.
+       """
+       session_factory = session_factory_from_settings(settings)
+       config = Configurator(root_factory=get_root, settings=settings)
+       config.set_session_factory(session_factory)
+       # ... other configuration stuff...
```

Comment:
Please see above example suggestion that uses a context manager and adapt accordingly.